packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.ts

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {
+    TreeWidget,
+    ContextMenuRenderer,
+    CompositeTreeNode,
+    ExpandableTreeNode,
+    SelectableTreeNode,
+    TreeModel,
+    TreeNode,
+    NodeProps,
+    LabelProvider,
+    TreeExpansionService,
+    ApplicationShell,
+    DiffUris
+} from "@theia/core/lib/browser";
+import { SearchInWorkspaceResult, SearchInWorkspaceOptions } from "../common/search-in-workspace-interface";
+import { SearchInWorkspaceService } from "./search-in-workspace-service";
+import { TreeProps } from "@theia/core/lib/browser";
+import { EditorManager, EditorDecoration, TrackedRangeStickiness, OverviewRulerLane, EditorWidget, ReplaceOperation } from "@theia/editor/lib/browser";
+import { inject, injectable } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { Path, CancellationTokenSource, Emitter, Event } from "@theia/core";
+import { WorkspaceService } from "@theia/workspace/lib/browser";
+import { h } from "@phosphor/virtualdom";
+import { MEMORY_TEXT } from "./memory-text-resource";
+import { FileResourceResolver } from "@theia/filesystem/lib/browser";
+
+export interface ResultNode extends ExpandableTreeNode, SelectableTreeNode {
```

Comment:
What about putting supplementary types under `SearchInWorkspaceResultTreeWidget`. `ResultNode`, `ResultLineNode` sound generic. please review for other generic top level concepts

Reply:
ok